### PR TITLE
refactor: Restrict binaryName types

### PR DESCRIPTION
**Pull request by elrrrrrrr**
> restrict binaryName type , the single source is the `config/binary.js` file.

* export `BinaryName` & `CategoryName` type
* use `BinaryNameRule` typebox validator in controller
* `binaryName: string` => `binaryName: BinaryName`

### CODE REVIEW

**Comment by killagu** [comment]
LGTM

File: app/port/controller/BinarySyncController.ts [hunk 1]
```diff
@@ -100,7 +103,13 @@ export class BinarySyncController extends AbstractController {
     path: '/-/binary/:binaryName(@[^/]{1,220}\/[^/]{1,220}|[^@/]{1,220})',
     method: HTTPMethodEnum.GET,
   })
-  async showBinaryIndex(@Context() ctx: EggContext, @HTTPParam() binaryName: string) {
+  async showBinaryIndex(@Context() ctx: EggContext, @HTTPParam() binaryName: BinaryName) {
+    // check binaryName valid
+    try {
+      ctx.tValidate(BinaryNameRule, binaryName);
+    } catch (e) {
+      throw new NotFoundError(`Binary "${binaryName}" not found`);
```
Comment (fengmk2) [inline app/port/controller/BinarySyncController.ts:111]: 没有加单测覆盖

File: config/binaries.ts [hunk 2]
```diff
@@ -879,6 +873,14 @@ const binaries: {
       },
     },
   },
+} as const;
+
+export type BinaryName = keyof typeof binaries;
```
Comment (fengmk2) [inline config/binaries.ts:878]: 学习了。

**Comment by fengmk2** [comment]
refactor label 不会自动发版本，需要设置为 feat

**Pull request merged**